app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -168,33 +180,132 @@ class PopupView : FrameLayout {
+                android.R.layout.simple_spinner_dropdown_item,
+                codedValuesNames
+            )
+            val codedValuePosition = codedValueDomain.codedValues.indexOfFirst { codedValue ->
+                codedValue.code == popupManager.getFieldValue(popupField)
+            }
+            // set the PopupField value as selected in the spinner
+            popupFieldCvdValues.setSelection(codedValuePosition)
+            popupFieldCvdValues.onItemSelectedListener =
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onNothingSelected(parent: AdapterView<*>?) {
+                        TODO("Not yet implemented")
+                    }
+
+                    override fun onItemSelected(
+                        parent: AdapterView<*>?,
+                        view: View?,
+                        position: Int,
+                        id: Long
+                    ) {
+                        popupManager.updateValue(
+                            codedValueDomain.codedValues[position].code,
+                            popupField
+                        )
+                    }
+                }
+        }
+
+        /**
+         * Updates the value of the specified PopupField to the appropriately cast string value of
+         * the specified value
+         */
+        private fun updateValue(popupField: PopupField, newValue: String): ArcGISRuntimeException? {
+            var error: ArcGISRuntimeException? = null
+            when (popupManager.getFieldType(popupField)) {
+                Field.Type.SHORT -> error =
+                    if (newValue.toShortOrNull() != null) {
+                        popupManager.updateValue(newValue.toShort(), popupField)
+                    } else {
+                        popupManager.updateValue(newValue, popupField)
+                    }
+                Field.Type.INTEGER -> error =
+                    if (newValue.toIntOrNull() != null) {
+                        popupManager.updateValue(newValue.toInt(), popupField)
+                    } else {
+                        popupManager.updateValue(newValue, popupField)
+                    }
+                Field.Type.FLOAT -> error =
+                    if (newValue.toFloatOrNull() != null) {
+                        popupManager.updateValue(newValue.toFloat(), popupField)
+                    } else {
+                        popupManager.updateValue(newValue, popupField)
+                    }
+                Field.Type.DOUBLE -> error =
+                    if (newValue.toDoubleOrNull() != null) {
+                        popupManager.updateValue(newValue.toDouble(), popupField)
+                    } else {
+                        popupManager.updateValue(newValue, popupField)
+                    }
+                Field.Type.TEXT -> error = popupManager.updateValue(newValue, popupField)
+
+                else -> Log.i(
+                    "PopupView",
```

Comment:
Should define a const TAG for this for re-use.

Reply:
defined the constant